ml_tools/model/gbm_strategy.py

```diff
@@ -372,31 +372,30 @@ def save_model(self, file_name: str) -> None:
             h5_file.create_dataset('serialized_lgbm_file', data=file_data)
 
 
-    def load_model(self, file_name: str) -> None:
+    def load_model(self, h5_file: h5py.File) -> None:
         """ A method for loading a trained model
 
         Parameters
         ----------
         file_name : str
             The name of the file to load the model from
         """
+        file_name = h5_file.filename
         lgbm_name = file_name.removesuffix(".h5") + ".lgbm" if file_name.endswith(".h5") else file_name + ".lgbm"
-        file_name = file_name if file_name.endswith(".h5") else file_name + ".h5"
 
         assert os.path.exists(file_name), f"file name = {file_name}"
```

Comment:
I don't think this assertion is necessary

Reply:
Fixed